op-chain-ops/immutables/immutables.go

```diff
@@ -1,63 +1,116 @@
-type ImmutableConfig map[string]ImmutableValues
-
-var Create2DeployerCodeHash = common.HexToHash("0xb0550b5b431e30d38000efb7107aaa0ade03d48a7198a140edda9d27134468b2")
-
-// Check does a sanity check that the specific values that
-// Optimism uses are set inside of the ImmutableConfig.
-func (i ImmutableConfig) Check() error {
-	if _, ok := i["L2CrossDomainMessenger"]["otherMessenger"]; !ok {
-		return errors.New("L2CrossDomainMessenger otherMessenger not set")
+// PredeploysImmutableConfig represents the set of L2 predeploys. It includes all
+// L2 predeploys - not just ones with immutable values. This is to be very explicit
+// about the configuration of the predeploys. It is important that the inner struct
+// fields are in the same order as the constructor arguments in the solidity code.
+type PredeploysImmutableConfig struct {
+	L2ToL1MessagePasser    struct{}
+	DeployerWhitelist      struct{}
+	WETH9                  struct{}
+	L2CrossDomainMessenger struct {
+		OtherMessenger common.Address
 	}
-	if _, ok := i["L2StandardBridge"]["otherBridge"]; !ok {
-		return errors.New("L2StandardBridge otherBridge not set")
+	L2StandardBridge struct {
+		OtherBridge common.Address
+		Messenger   common.Address
 	}
-	if _, ok := i["L2ERC721Bridge"]["messenger"]; !ok {
-		return errors.New("L2ERC721Bridge messenger not set")
+	SequencerFeeVault struct {
+		Recipient           common.Address
+		MinWithdrawalAmount *big.Int
+		WithdrawalNetwork   uint8
 	}
-	if _, ok := i["L2ERC721Bridge"]["otherBridge"]; !ok {
-		return errors.New("L2ERC721Bridge otherBridge not set")
+	OptimismMintableERC20Factory struct {
+		Bridge common.Address
 	}
-	if _, ok := i["OptimismMintableERC721Factory"]["bridge"]; !ok {
-		return errors.New("OptimismMintableERC20Factory bridge not set")
+	L1BlockNumber       struct{}
+	GasPriceOracle      struct{}
+	L1Block             struct{}
+	GovernanceToken     struct{}
+	LegacyMessagePasser struct{}
+	L2ERC721Bridge      struct {
+		Messenger   common.Address
+		OtherBridge common.Address
 	}
-	if _, ok := i["OptimismMintableERC721Factory"]["remoteChainId"]; !ok {
-		return errors.New("OptimismMintableERC20Factory remoteChainId not set")
+	OptimismMintableERC721Factory struct {
+		Bridge        common.Address
+		RemoteChainId *big.Int
 	}
-	if _, ok := i["SequencerFeeVault"]["recipient"]; !ok {
-		return errors.New("SequencerFeeVault recipient not set")
+	ProxyAdmin   struct{}
+	BaseFeeVault struct {
+		Recipient           common.Address
+		MinWithdrawalAmount *big.Int
+		WithdrawalNetwork   uint8
 	}
-	if _, ok := i["L1FeeVault"]["recipient"]; !ok {
-		return errors.New("L1FeeVault recipient not set")
+	L1FeeVault struct {
+		Recipient           common.Address
+		MinWithdrawalAmount *big.Int
+		WithdrawalNetwork   uint8
 	}
-	if _, ok := i["BaseFeeVault"]["recipient"]; !ok {
-		return errors.New("BaseFeeVault recipient not set")
+	SchemaRegistry struct{}
+	EAS            struct {
+		Name string
+	}
+	Create2Deployer struct{}
```

Comment:
This means that Arachnid's create2 deployer is already being included as a predeploy for all OP chains? So I don't need to include it in #8281?

Reply:
The canyon hardfork adds the create2 deployer via the state transition function. Its a hack but necessary because of a nonce issue that happened on base

Reply:
I would double check the address and if they are the same address then you do not need to include it